### PR TITLE
Update to GAX v3, fixing snippets accordingly

### PR DESCRIPTION
**Pull request by jskeet**
(Will assign when I've re-reviewed myself.)

### CODE REVIEW

File: Google.Api.Generator/Generation/SnippetCodeGenerator.cs [hunk 1]
```diff
@@ -435,8 +435,8 @@ private IEnumerable<ObjectInitExpr> InitRequest()
                         makeRequest,
                         BlankLine,
                         "// Read streaming responses from server until complete",
-                        ResponseStream.WithInitializer(Response.Access(nameof(ServerStreamingBase<int>.ResponseStream))),
-                        While(Await(ResponseStream.Call(nameof(IAsyncEnumerator<int>.MoveNext))()))(
+                        ResponseStream.WithInitializer(Response.Call(nameof(ServerStreamingBase<int>.GetResponseStream))(Ctx.Type(typeof(CancellationToken)).Access(nameof(CancellationToken.None)))),
```
Comment (chrisdunelm) [inline Google.Api.Generator/Generation/SnippetCodeGenerator.cs]: I think there's a slight shortcut you can use: `Ctx.Type<CancellationToken>()`

File: Google.Api.Generator/Generation/SnippetCodeGenerator.cs [hunk 2]
```diff
@@ -484,8 +484,8 @@ private IEnumerable<ObjectInitExpr> InitRequest()
                         BlankLine,
                         "// Create task to do something with responses from server",
                         ResponseHandlerTask.WithInitializer(Ctx.Type<Task>().Call(nameof(Task.Run))(LambdaTyped(null, async: true)(
-                            ResponseStream.WithInitializer(Response.Access(nameof(ServerStreamingBase<int>.ResponseStream))),
-                            While(Await(ResponseStream.Call(nameof(IAsyncEnumerator<int>.MoveNext))()))(
+                            ResponseStream.WithInitializer(Response.Call(nameof(ServerStreamingBase<int>.GetResponseStream))(Ctx.Type(typeof(CancellationToken)).Access(nameof(CancellationToken.None)))),
```
Comment (chrisdunelm) [inline Google.Api.Generator/Generation/SnippetCodeGenerator.cs]: (Ditto here)

File: Google.Api.Generator/Generation/SnippetCodeGenerator.cs (hunk 1 above)
Comment (jskeet) [reply Google.Api.Generator/Generation/SnippetCodeGenerator.cs]: Ooh thanks, will do that.

**Comment by jskeet** [comment]
Will merge on green. There's an annoyance where the first commit seems to move a submodule backwards, with a later one moving it forward - but I can't work out how to fix that.

**Pull request merged**